cirq/circuits/circuit.py

```diff
@@ -1228,6 +1228,25 @@ def all_operations(self) -> Iterator[ops.Operation]:
         """
         return (op for moment in self for op in moment.operations)
 
+    def max_qid_shape(
```

Comment:
This method needs a docstring.

Could you describe why we need this method, instead of just a shape method that takes a qid_order?

The `qid_min` argument seems unnecessary, since `np.max` exists.

Reply:
Because circuits don't really have a well-defined qubit order but everything else does.

Something like this?  `tuple(np.max([(1,2,3,4),(2,)*4], axis=0)) => array([2, 2, 3, 4])`